ext/standard/tests/strings/chunk_split_variation1_32bit.phpt

```diff
@@ -0,0 +1,28 @@
+--TEST--
+Test chunk_split() function : usage variations - unexpected large '$end' string argument variation 1
+--SKIPIF--
+<?php
+if (PHP_INT_SIZE != 4) die("skip this test is for 32bit platform only");
+?>
+--FILE--
+<?php
+/* Prototype  : string chunk_split(string $str [, int $chunklen [, string $ending]])
+ * Description: Returns split line
+ * Source code: ext/standard/string.c
+ * Alias to functions: none
+*/
+
+echo "*** Testing chunk_split() : unexpected large 'end' string argument variation 1 ***\n";
+
+$a=str_repeat("B", 65535);
+$b=1;
+$c=str_repeat("B", 65535);
+var_dump(chunk_split($a,$b,$c));
+?>
+--EXPECTF--
+*** Testing chunk_split() : unexpected large 'end' string argument variation 1 ***
+mmap() failed: [12] Cannot allocate memory
+
+mmap() failed: [12] Cannot allocate memory
```

Comment:
Heh, looks like we have a bug in ZMM... This should have resulted in a memory_limit error, not an attempt to perform a large allocation. From a quick look, the memory limit checks themselves are subject to overflow right now...

Reply:
Well uncovering some hidden bugs it seems like :D

Reply:
Should be fixed by https://github.com/php/php-src/commit/16d35eb643bf974554e5264021ee10fc969e2053.